### PR TITLE
Server testing

### LINKED ISSUE #17: Setup Continuous Integration
## User Story
As a **user**, I would like **tests to be run automatically and the game to be deployed only if all tests pass** so that **I will not have to play a buggy game**.

## Detailed Description
Research the different continuous deployment tools: Heroku/Travis/Git Actions. Find one that is easy to setup and maintain, and is free to host the backend. After this, add a sample test case and setup continuous integration using the CICD tool researched.

## Acceptance Criteria
- [x] Given **a choice of game engine**, then **we know which tool is easiest and free for the backend**
- [x] Given **a choice of backend framework**, then **we have a corresponding testing framework set up to run tests**
- [ ] Given **a test fails**, then **code is not pushed automatically**

### PR DESCRIPTION
**Pull request by ghuser1023**
Adds testing via Mocha (https://mochajs.org/), akin to what's in https://socket.io/docs/v3/testing/. This enables us to:
- Write integration tests that involve actual messages sent over sockets, like in `tests\server.test.ts` below
- Write unit tests that test individual functionality (not yet defined)

This was chosen over mocking via https://www.npmjs.com/package/socket.io-mock for the following reasons:
- We don't actually want to mock out the sockets, since we want to make sure that they're working.
- Mocking is complex under the hood (not reflected in the simplicity of the provided example), and hurts my brain.

Changes of note:
- Moved all server logic from `index.ts` -> `server.ts`, leaving just the initialization in `index.ts`. This lets us write test code that tests the actual server implementation.
- A simple test (under `tests/`) testing the current server functionality -- this should be rewritten once the server starts to actually do more things.
- New autodoc generation command, reflecting the existence of `server.ts`.

ADDITIONAL:
- Update heroku api key to allow continous deployment to work properly

### CODE REVIEW

File: server/README.md [hunk 1]
```diff
@@ -23,9 +23,13 @@
     - Currently, RemoteController.cs is attached to the RemoteController GameObject
```
Comment (reginawang3495) [inline server/README.md:26]: at like line 17 we can change to server.ts if you want, ow i can do on my own next commit!

**Review by reginawang3495 — DISMISSED** [review]
noice :)

**Comment by reginawang3495** [comment]
oh also do we wanna maybe add a setup testing task to #17 or something?

**Comment by ghuser1023** [comment]
> at like line 17 we can change to server.ts if you want, ow i can do on my own next commit!

fixed

> oh also do we wanna maybe add a setup testing task to #17 or something?

added a bullet point! don't quite feel like making a task haha

**Review by reginawang3495 — APPROVED** [review]
resolved conflicts and update api key for CD!

**Pull request merged**